compliance/collection/nodeinventorizer/nodeinventory_test.go

```diff
@@ -0,0 +1,211 @@
+package nodeinventorizer
+
+import (
+	"testing"
+
+	"github.com/stackrox/scanner/database"
+	scannerV1 "github.com/stackrox/scanner/generated/scanner/api/v1"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestNodeInventorizer(t *testing.T) {
+	suite.Run(t, new(NodeInventorizerTestSuite))
+}
+
+type NodeInventorizerTestSuite struct {
+	suite.Suite
+}
+
+func (s *NodeInventorizerTestSuite) TestConvertRHELComponentIDs() {
+	testCases := map[string]struct {
+		inComponents  []*database.RHELv2Package
+		outComponents []*scannerV1.RHELComponent
+		expectedLen   int
+	}{
+		"nil-inComponents": {
+			inComponents: nil,
+		},
+		"one-component": {
+			inComponents: []*database.RHELv2Package{
+				{
+					Name:    "zlib",
+					Version: "1.2.11-16.el8_2",
+					Arch:    "x86_64",
+					ExecutableToDependencies: database.StringToStringsMap{
+						"/usr/lib64/libz.so.1":      {},
+						"/usr/lib64/libz.so.1.2.11": {},
+					},
+				},
+			},
+			expectedLen: 1,
+		},
+		"multi-component": {
+			inComponents: []*database.RHELv2Package{
+				{
+					Name:    "zlib",
+					Version: "1.2.11-16.el8_2",
+					Arch:    "x86_64",
+					ExecutableToDependencies: database.StringToStringsMap{
+						"/usr/lib64/libz.so.1":      {},
+						"/usr/lib64/libz.so.1.2.11": {},
+					},
+				},
+				{
+					Name:    "redhat-release",
+					Version: "8.3-1.0.el8",
+					Arch:    "x86_64",
+				},
+			},
+			expectedLen: 2,
+		},
+		"collision-component": {
+			inComponents: []*database.RHELv2Package{
+				{
+					Name:    "redhat-release",
+					Version: "8.3-1.0.el8",
+					Arch:    "x86_64",
+				},
+				{
+					Name:    "redhat-release",
+					Version: "8.3-1.0.el8",
+					Arch:    "x86_64",
+				},
+			},
+			expectedLen: 1,
+		},
+	}
+	for caseName, testCase := range testCases {
+		s.Run(caseName, func() {
+			mockComponents := &database.RHELv2Components{
+				Dist:     "MockDist",
+				CPEs:     nil,
+				Packages: testCase.inComponents,
+			}
+			convertedComponents := convertRHELComponents(mockComponents)
+			if testCase.inComponents != nil {
+				convertedIDs := make([]int64, 0, len(convertedComponents))
```

Comment:
Why do we need `convertedIDs`? I cannot see it being used anywhere after it is filled